File: Semaine_2/Abacus/Abacus/ViewController.m
//
//  ViewController.m
//  Dizainier_
//
//  Created by Robin Blottière--Mayo on 05/09/2018.
//  Copyright © 2018 Robin Blottière--Mayo. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
	[super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
}


- (void)didReceiveMemoryWarning {
	[super didReceiveMemoryWarning];
	// Dispose of any resources that can be recreated.
}

- (void)assignmentMethode:(int)value {
	NSString *decimalNumber;
	if (_SwitchOutlet.isOn) {
		decimalNumber = [NSString stringWithFormat:@"%X", value];
	} else {
		decimalNumber = [NSString stringWithFormat:@"%d", value];
	}
	if ((int)_StepperOutlet.value == 42) {
		[_NumberLabelOutlet setTextColor:UIColor.redColor];
		[_NumberLabelOutlet setFont:[UIFont boldSystemFontOfSize:17.0]];
	} else {
		[_NumberLabelOutlet setTextColor:UIColor.whiteColor];
		[_NumberLabelOutlet setFont:[UIFont systemFontOfSize:17.0]];
	}
	[_NumberLabelOutlet setText:decimalNumber];
	[_SliderOutlet setValue:value];
	_TensSegmentOutlet.selectedSegmentIndex = value / 10;
	_UnitsSegmentOutlet.selectedSegmentIndex = value % 10;
	[decimalNumber release];
}

- (IBAction)StepperAction:(id)sender {
	[self assignmentMethode:(int)_StepperOutlet.value];
}

- (IBAction)SliderAction:(id)sender {
	_StepperOutlet.value = (int)_SliderOutlet.value;
	[self assignmentMethode:(int)_StepperOutlet.value];
}

- (void)segmentedControlMethode {
	_StepperOutlet.value = (int)_UnitsSegmentOutlet.selectedSegmentIndex + 10 * (int)_TensSegmentOutlet.selectedSegmentIndex;
	[self assignmentMethode:(int)_StepperOutlet.value];
}

- (IBAction)TensSegmentAction:(id)sender {
	[self segmentedControlMethode];
}

- (IBAction)UnitsSegmentAction:(id)sender {
	[self segmentedControlMethode];
}

- (IBAction)SwitchAction:(id)sender {
	NSString *decimalNumber;
	if (_SwitchOutlet.isOn) {
		decimalNumber = [NSString stringWithFormat:@"%X", (int)_StepperOutlet.value];
	} else {
		decimalNumber = [NSString stringWithFormat:@"%d", (int)_StepperOutlet.value];
	}
	[_NumberLabelOutlet setText:decimalNumber];
	[decimalNumber release];
}

- (IBAction)RaZButton:(UIButton *)sender {
	_StepperOutlet.value = 0;
	[_NumberLabelOutlet setText:@"0"];
	[_SliderOutlet setValue:0];
	_TensSegmentOutlet.selectedSegmentIndex = 0;
	_UnitsSegmentOutlet.selectedSegmentIndex = 0;
}

@end

File: Semaine_3/Abacus_S/Abacus_S/ViewController.swift
//
//  ViewController.swift
//  Dizainier_S
//
//  Created by Robin Blottière--Mayo on 23/09/2018.
//  Copyright © 2018 Robin Blottière--Mayo. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

	override func viewDidLoad() {
		super.viewDidLoad()
		// Do any additional setup after loading the view, typically from a nib.
	}

	@IBOutlet weak var NumberLabelOutlet: UILabel!
	@IBOutlet weak var StepperOutlet: UIStepper!
	@IBOutlet weak var SliderOutlet: UISlider!
	@IBOutlet weak var TensSegmentOutlet: UISegmentedControl!
	@IBOutlet weak var UnitsSegmentOutlet: UISegmentedControl!
	@IBOutlet weak var SwitchOutlet: UISwitch!

	func assignmentMethode(value : Int) {
		var decimalNumber : String
		if (SwitchOutlet.isOn) {
			decimalNumber = String(value, radix:16)
		} else {
			decimalNumber = String(value)
		}
		if value == 42 {
			NumberLabelOutlet.textColor = UIColor.red
			NumberLabelOutlet.font = UIFont.boldSystemFont(ofSize: 17)
		} else {
			NumberLabelOutlet.textColor = UIColor.white
			NumberLabelOutlet.font = UIFont.systemFont(ofSize: 17)
		}
		NumberLabelOutlet.text = decimalNumber
		SliderOutlet.value = Float(value)
		TensSegmentOutlet.selectedSegmentIndex = Int(value / 10)
		UnitsSegmentOutlet.selectedSegmentIndex = Int(value % 10)
	}
	
	@IBAction func StepperAction(_ sender: UIStepper) {
		assignmentMethode(value:Int(StepperOutlet.value))
	}
	
	@IBAction func SliderAction(_ sender: UISlider) {
		StepperOutlet.value = Double(SliderOutlet.value)
		assignmentMethode(value:Int(StepperOutlet.value))
	}
	
	@IBAction func segmentAction(_ sender: UISegmentedControl) {
		StepperOutlet.value = Double(UnitsSegmentOutlet.selectedSegmentIndex + 10 * TensSegmentOutlet.selectedSegmentIndex)
		assignmentMethode(value:Int(StepperOutlet.value))	}
	
	@IBAction func SwitchAction(_ sender: UISwitch) {
		var decimalNumber : String
		if (SwitchOutlet.isOn) {
			decimalNumber = String(Int(StepperOutlet.value), radix:16)
		} else {
			decimalNumber = String(Int(StepperOutlet.value))
		}
		NumberLabelOutlet.text = decimalNumber
	}
	
	@IBAction func RaZAction(_ sender: UIButton) {
		if StepperOutlet.value == 42 {
			NumberLabelOutlet.textColor = UIColor.white
			NumberLabelOutlet.font = UIFont.systemFont(ofSize: 17)
		}
		StepperOutlet.value = 0
		SliderOutlet.value = 0
		UnitsSegmentOutlet.selectedSegmentIndex = 0
		TensSegmentOutlet.selectedSegmentIndex = 0
		NumberLabelOutlet.text = "0"
	}
}

